test/src/code_field/code_controller_comment_uncomment_test.dart

```diff
@@ -4,250 +4,506 @@ import 'package:flutter/material.dart';
+            expectedFullText: '''
+a
+''',
+            expectedSelection: TextSelection(baseOffset: 0, extentOffset: 2),
+          ),
+          _Example(
+            'WHEN line is commented out with the second type of comment '
+            'SHOULD uncomment it',
+            initialFullText: '''
+# a
+''',
+            initialSelection: TextSelection.collapsed(offset: 0),
+            expectedFullText: '''
+a
+''',
+            expectedSelection: TextSelection(baseOffset: 0, extentOffset: 2),
+          ),
+          _Example(
+            'WHEN line is empty '
+            'SHOULD only change selection',
+            initialFullText: '''
+
+''',
+            initialSelection: TextSelection.collapsed(offset: 0),
+            expectedFullText: '''
+
+''',
+            expectedSelection: TextSelection(baseOffset: 0, extentOffset: 1),
+          ),
+          _Example(
+            'WHEN empty line between commented lines is selected '
+            'SHOULD only change selection',
+            initialFullText: '''
+// a
+
+# a
+''',
+            initialSelection: TextSelection.collapsed(offset: 5),
+            expectedFullText: '''
+// a
+
+# a
+''',
+            expectedSelection: TextSelection(baseOffset: 5, extentOffset: 6),
+          ),
+        ];
+
+        for (final example in examples) {
+          final controller = CodeController(language: php);
+          controller.text = example.initialFullText;
+          controller.selection = example.initialSelection;
+
+          controller.commentOutOrUncommentSelection();
+
+          expect(
+            controller.value.text,
+            example.expectedFullText,
+            reason: example.name,
+          );
+          expect(
+            controller.code.text,
+            example.expectedFullText,
+            reason: example.name,
+          );
+          expect(
+            controller.value.selection,
+            example.expectedSelection,
+            reason: example.name,
+          );
+        }
+      });
+      test('Selection is a range', () {
+        final examples = [
+          _Example(
+            'WHEN every selected line has different type of single line comments '
+            'SHOULD uncomment all of them',
+            initialFullText: '''
+// aA
+# AA
+// AA
+# Aa
+''',
+            initialSelection: TextSelection(baseOffset: 4, extentOffset: 19),
+            expectedFullText: '''
+aA
+AA
+AA
+Aa
+''',
+            expectedSelection: TextSelection(baseOffset: 0, extentOffset: 12),
+          ),
+          _Example(
+            'WHEN every selected line has different type of single line comments '
+            'and one line inbetween is empty '
+            'SHOULD uncomment all commented lines and do not affect empty line',
+            initialFullText: '''
+// aA
+# AA
+
+# Aa
+''',
+            initialSelection: TextSelection(baseOffset: 4, extentOffset: 15),
+            expectedFullText: '''
+aA
+AA
+
+Aa
+''',
```

Comment:
That's expected. 
If none of the lines are uncommented (all of them are comments) we remove uncomment the lines.

And it will not cause to return false.
false is returned only if every line returns false.
If there is `a` line that contains text but is not a comment => true